dnsapi/dns_nsupdate.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+
+########  Public functions #####################
+
+#Usage: dns_nsupdate_add   _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_nsupdate_add() {
+  fulldomain=$1
+  txtvalue=$2
+  _checkKeyFile || return 1
+  NSUPDATE_SERVER=${NSUPDATE_SERVER:-localhost}
+  # save the dns server and key to the account conf file.
+  _saveaccountconf NSUPDATE_SERVER "${NSUPDATE_SERVER}"
+  _saveaccountconf NSUPDATE_KEY "${NSUPDATE_KEY}"
+  tmp=$(mktemp --tmpdir acme_nsupdate.XXXXXX)
```

Comment:
Use our internal function `_mktemp()`  instead.
In some platforms, the command `mktemp` is not available.
And the command syntax for `mktemp` is different between GnuLinux and BSD/Mac.